datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTable.java

```diff
@@ -642,6 +642,25 @@ public void print(StringBuilder appendable) {
         printer.printTable(raw, appendable);
     }
 
+    /**
+     * Prints a string representation of this
+     * table to the {@code appendable},
+     * with custom indentation
+     * and without escaping cell contents
+     *
+     * @param appendable to append the string representation
+     *                   of this table to.
+     * @param indent     level of start indentation
+     * @throws IOException
+     */
+    public void printFormatted(Appendable appendable, String indent) throws IOException {
```

Comment:
This isn't scaling very well. For each different way to print we'd end up with a new method. Might be better to have the `TablePrinter` as part of the public API and drop the `print*` methods from the data table. 

The table printer could take the `DataTable` as an argument and use `.cells()` to get the cells.

I can try to do this or you can if you want.

Reply:
Just pushed an update, if that's more what you had in mind? Am available next couple of days to fix it up further, but equally my ego won't be bruised if you prefer to make some changes directly :)

Reply:
(also removed the printer-builder class from the last commit, and allowed for configuring the printer class directly, to minimise number of files, but I don't feel strongly about it)

Reply:
Getting close!

The idea is to use the builder pattern to create immutable table printer that can be used to print many different tables. Using the builder pattern to create immutable objects does take a bit more code but makes the objects much easier to work with. Especially in multi-threaded contexts.

So we have a `TablePrinter` with these public methods:

````
TablePrinter
 - static TablePrinter.Builder builder()
 - printTable(DataTable table, StringBuilder appendable)
 - printTable(DataTable table, Appendable appendable)
```

And a `TablePrinter.Builder` with these public methods:

```
TablePrinter.Builder
 - indent(int)
 - escape(boolean)
 - TablePrinter build()
```

I realize this may not be apparent from the surrounding code but a lot of this code was written in a time before these ideas were common.

Reply:
cool yeah, that makes sense.

Struggled how best to handle the child `DiffTablePrinter` class, whether it should have its own builder, or call the parent constructor with the parent's builder (for now went with the latter)

Reply:
I had a look. Replaced inheritance with composition by extracting the `indent` as a function that returns a specific indent prefix for each row.